analyzer/src/main/kotlin/curation/Sw360PackageCurationProvider.kt

```diff
@@ -0,0 +1,101 @@
+
+
+class Sw360PackageCurationProvider(sw360Configuration: Sw360Configuration?) : PackageCurationProvider {
+
+    private val sw360ConnectionConfig: SW360ClientConfig = SW360ClientConfig.createConfig(
+        sw360Configuration?.restUrl,
+        sw360Configuration?.authUrl,
+        sw360Configuration?.username,
+        sw360Configuration?.password,
+        sw360Configuration?.clientId,
+        sw360Configuration?.clientPassword,
+        ServiceFactory().createHttpClient(
+            sw360Configuration?.proxyHost.isNullOrEmpty() && sw360Configuration?.proxyPort!! > 0,
+            sw360Configuration?.proxyHost,
+            sw360Configuration?.proxyPort!!
+        ),
+        ServiceFactory.getObjectMapper()
+    )
+
+    private val sw360ConnectionFactory = SW360ConnectionFactory().newConnection(sw360ConnectionConfig)
+    private val sw360ReleaseClient = sw360ConnectionFactory.releaseAdapter
+
+    override fun getCurationsFor(pkgId: Identifier): List<PackageCuration> {
+        val name = listOfNotNull(pkgId.namespace, pkgId.name).joinToString("/")
+
+        return sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkgId.version).takeIf { it.isPresent }?.let {
+            val curation = it.get()
+            val attachmentsDownloadPath = createTempDir()
+
+            listOf(
+                PackageCuration(
+                    id = pkgId,
+                    data = PackageCurationData(
+                        declaredLicenses = curation.mainLicenseIds?.toSortedSet() ?: sortedSetOf(),
+                        homepageUrl = "", // Not provided from SW360.
+                        sourceArtifact = getSourceArtifactFromRelease(curation, attachmentsDownloadPath),
+                        vcs = null,
+                        comment = "Provided by SW360."
+                    )
+                )
+            )
+        } ?: emptyList()
+    }
+
+    private fun getSourceArtifactFromRelease(release: SW360SparseRelease, downloadPath: File) : RemoteArtifact? =
+        sw360ReleaseClient.getReleaseById(release.releaseId).takeIf { it.isPresent }?.let { it ->
+            val fullRelease = it.get()
+            val attachments = fullRelease.embedded?.attachments
+            attachments?.filter { it.attachmentType == SW360AttachmentType.SOURCE }
+                ?.map { attachment ->
+                    sw360ReleaseClient.downloadAttachment(fullRelease, attachment, downloadPath.toPath())
+                        .takeIf { it.isPresent }.let {
+                            return RemoteArtifact(
+                                url = it.toString(),
+                                hash = Hash(
+                                    value = attachment.sha1,
+                                    algorithm = HashAlgorithm.SHA1
+                                )
+                            )
+                        }
+                }
+            RemoteArtifact.EMPTY
+        }
```

Comment:
Having a `return` inside an expression function is a bit unusual and long expression functions can become hard to understand. I believe making this not an expression function and using `if` instead of `takeIf` could make this function easier to read.